### PR TITLE
[NO-ISSUE] feat(users): allow to provide external userId (uuid)

### PR DESCRIPTION
**Pull request by GuillaumeDecMeetsMore**
### Changed
- Added possibility of providing the user's UUID (PK) when creating a user

### CODE REVIEW

File: scheduler/clients/javascript/lib/userClient.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ type UpdateUserRequest = {
 
 type CreateUserRequest = {
   metadata?: Metadata;
+  userId?: string;
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/clients/javascript/lib/userClient.ts:23]: Will add JSDocs to all client lib's exposed functions/types in another PR

File: scheduler/crates/api/src/user/create_user.rs
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/user/create_user.rs:1]: Main change (1)

File: scheduler/crates/api_structs/src/user/api.rs
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api_structs/src/user/api.rs:1]: Main change (2)

File: scheduler/crates/domain/src/user.rs
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/domain/src/user.rs:1]: Main change (3)

**Review by mm-zacharydavison — APPROVED** [review]

**Review by mm-zacharydavison — APPROVED** [review]

**Review by mm-zacharydavison — APPROVED** [review]

**Pull request merged**